core/src/test/scala/caliban/TestUtils.scala

```diff
@@ -563,4 +564,11 @@ object TestUtils {
 
     val resolverEmpty = new RootResolver(Option.empty[Unit], Option.empty[Unit], Option.empty[Unit])
   }
+
+  val skipJdk8: TestAspectPoly = {
+    val jdkVersionArr = System.getProperty("java.version").split('.')
+    val maybeMajor    = jdkVersionArr.head.toInt
```

Comment:
Use `headOption` so that it doesn't throw here. I think by default we just run the test if we can't identity the version.

Reply:
Good point. Changed it to pattern matching instead since it makes the code a bit more clear I think